developing-packages.md

```diff
@@ -0,0 +1,174 @@
+* `ios/Classes/HelloPlugin.m`: 
+   - The iOS platform specific implementation of the plugin API.
+* `example/`:
+   - A Flutter app that depends on the plugin, and illustrates how to use it.
+
+By default, the plugin project uses Objective-C for iOS code and
+Java for Android code. If you prefer Swift or Kotlin, you can specify the
+iOS language using `-i` and/or the Android language using `-a`. For example:
+```
+flutter create --plugin -i swift -a kotlin hello
+```
+
+### Editing plugin source code {#edit-code}
+
+As plugin code involves several platforms and programming languages, some
+specific steps are needed to ensure a smooth experience.
+
+#### Plugin API code (.dart)
+
+To edit the Dart plugin API code, open `hello/` in IntelliJ IDEA (or your
+favorite Dart editor). The plugin API is located in `lib/hello.dart` shown in
+the Project view.
+
+To run the plugin, you need to launch the plugin **example app**, which requires
+defining a launch configuration:
+
+1. Select 'Run > Edit Configurations...'.
+1. Select '+' and then 'Flutter'.
+1. In 'Dart entrypoint', enter `<plugin folder>/example/lib/main.dart`.
+1. Select 'OK'.
+1. Launch the example app with 'Run' or 'Debug'.
+
+*Note*: [Hot reload](https://flutter.io/faq/#hot-reload) is only supported for
+changes to Dart code, not for Android and iOS code.
+
+#### Android platform code (.java/.kt)
+
+Before editing the Android platform code in Android Studio, first make sure that
+the code has been built at least once (i.e., run the example app from IntelliJ, 
+or in a terminal execute `cd hello/example; flutter build apk`). 
+
+Next,
+
+1. Launch Android Studio
+1. Select 'Import project' in 'Welcome to Android Studio' dialog, or select 
+'File > New > Import Project...'' in the menu, and select the
+`hello/example/android/build.gradle` file.
+1. In the 'Gradle Sync' dialog, select 'OK'.
+1. In the 'Android Gradle Plugin Update' dialog, select 'Don't remind me again for this project'. 
+
+The Android platform code of your plugin is located in `hello/java/com.yourcompany.hello/HelloPlugin`.
+
+You can run the example app from Android Studio by pressing the &#9654; button.
+
+#### iOS platform code (.h+.m/.swift)
+
+Before editing the iOS platform code in Xcode, first make sure that
+the code has been built at least once (i.e., run the example app from IntelliJ, 
+or in a terminal execute `cd hello/example; flutter build ios`).
+
+Next,
+
+1. Launch Xcode
+1. Select 'File > Open', and select the `hello/example/ios/Runner.xcworkspace` file.
+
+The iOS platform code of your plugin is located in `Pods/Development Pods/hello/Classes/` in the Project Navigator.
+
+You can run the example app by pressing the &#9654; button.
+
+   
+## Publishing packages {#publish}
+
+Once you have implemented a package, you can publish it on
+[Pub](https://pub.dartlang.org/), so that other developers can easily use it.
+
+Prior to publishing, make sure to complete the `pubspec.yaml`, `README.md`, and
```

Comment:
what does "complete" mean? Is there something they _must_ do before publishing? If so, let's tell them.